src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/haflow/HaFlowRuleManagerProcessingAction.java

```diff
@@ -86,4 +91,15 @@ protected HaFlow copyHaFlowWithPathIds(
         }
         return haFlow;
     }
+
+    protected Set<PathId> getOverlappingPathIds(HaFlow haFlow) {
```

Comment:
I don't get it. Let's say inside for loop I added path IDs [1,2,3,4], then I remove all that are present in subpathIds collection, which is say [1,2]. So I return [3,4], but why they are called overlapping? Seems the opposite: it gives non-overlapping, but only from one side.
I understand overlap as intersection of the sets, I'd expect to add to a set all elements of one collection that are at the same time present in the other using contains.

Reply:
It think it is bat naming again. Let me explain. We have 2 types of ingress rules.
1. own rule of the flow - 1 rule for each primary path of each flow
2. shared rules - these rules are shared by several flows (when flows share one in port for example)

When we removing/adding rules for some flow to the switch we must not remove shared rules which are used by other flows and we should not re install shared rules which are used by other flows

This method finds all paths (except own haFlow paths and protected paths) which have save source endpoint as haFlow endpoints. This pathIds will be used in RuleManager to check are they use shared rules or not.

Reply:
thanks for the explanation. So, a better name with the get semantics could be: unshared, independent, solo, or something similar. Or it could be rewritten into something like excludeOverlappingPaths

Reply:
thank you for review! fixed